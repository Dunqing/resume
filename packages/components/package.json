{
  "name": "@resumejs/components",
  "version": "0.9.5",
  "author": "dengqing <dengqing0821@gmail.com>",
  "license": "MIT",
  "homepage": "https://resume.todev.cc",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/Dunqing/resume.git"
  },
  "bugs": {
    "url": "https://github.com/Dunqing/resume/issues"
  },
  "keywords": [
    "resume",
    "components"
  ],
  "exports": {
    ".": {
      "types": "./dist/index.d.ts",
      "import": "./dist/index.mjs",
      "require": "./dist/index.js"
    }
  },
  "main": "dist/index.js",
  "module": "dist/index.mjs",
  "types": "dist/index.d.ts",
  "scripts": {
    "dev": "vite build --watch",
    "build": "vite build --emptyOutDir",
    "prepublishOnly": "pnpm run build"
  },
  "peerDependencies": {
    "react": "^17.0.2 || ^18.0.0",
    "react-dom": "^17.0.2 || ^18.0.0"
  },
  "dependencies": {
    "@resumejs/template": "workspace:*",
    "@resumejs/template-default": "workspace:*",
    "@types/hast": "^3.0.0",
    "clsx": "^2.0.0",
    "hast-util-has-property": "^3.0.0",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "react-markdown": "^8.0.7",
    "rehype-raw": "^6.1.1",
    "remark-frontmatter": "^4.0.1",
    "remark-gfm": "^3.0.1",
    "unified": "^11.0.0",
    "unist-builder": "^4.0.0",
    "unist-util-is": "^6.0.0",
    "unist-util-visit": "^5.0.0",
    "vite-plugin-dts": "^3.5.1",
    "yaml": "^2.3.1"
  },
  "devDependencies": {
    "@types/mdast": "^4.0.0",
    "@types/react": "^18.2.20",
    "@types/react-dom": "^18.2.7",
    "@vitejs/plugin-react": "^4.0.4",
    "vite": "^4.4.9"
  },
  "publishConfig": {
    "access": "public"
  }
}
