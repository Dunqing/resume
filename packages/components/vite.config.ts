import { defineConfig } from 'vite'
import react from '@vitejs/plugin-react'
import dts from 'vite-plugin-dts'
import importCss from 'vite-plugin-import-css'
import packageJson from './package.json'

// https://vitejs.dev/config/
export default defineConfig({
  plugins: [dts(), react(), importCss()],
  build: {
    lib: {
      entry: 'src/index.ts',
      formats: ['es', 'cjs'],
      fileName: (format) => `index.${format === 'cjs' ? 'js' : 'mjs'}`,
    },
    emptyOutDir: false,
    rollupOptions: {
      external: Object.keys(packageJson.dependencies),
    },
  },
})
