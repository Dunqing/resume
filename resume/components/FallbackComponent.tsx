import type { ReactNode } from 'react'
import React from 'react'
import type { FallbackProps } from 'react-error-boundary'

export const FallbackComponent: React.FC<FallbackProps> = (props) => {
  if (props.error) {
    return (
      <div className="flex justify-center ">
        <div className="max-w-5xl p-2 ">
          <h1 className="text-5xl font-extrabold text-red ">
            Please check your resume
          </h1>
          <p className="mt-10 text-2xl text-dark-300">{props.error.stack}</p>
        </div>
      </div>
    )
  }
  return <>{(props as unknown as { children: ReactNode }).children}</>
}
