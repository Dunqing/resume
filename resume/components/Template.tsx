import type { ChangeEvent } from 'react'
import { useTemplate } from '../contexts/template'

const templateNameList = ['default', 'nova']

const Template: React.FC = () => {
  const { template, setTemplate } = useTemplate()
  const onTemplate = (event: ChangeEvent<HTMLSelectElement>) => {
    const value = event.target.value
    setTemplate(value)
  }
  return (
    <label className="text-sm">
      切换模板：
      <select
        className="dark:bg-dark-400"
        onChange={onTemplate}
        defaultValue={template}
      >
        {templateNameList.map((name) => {
          return (
            <option
              value={name}
              key={name}
            >
              {name}
            </option>
          )
        })}
      </select>
    </label>
  )
}

export default Template
