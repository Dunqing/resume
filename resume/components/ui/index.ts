export * from './button'
export * from './hover-card'
export * from './label'
export * from './separator'
export * from './tabs'
export * from './textarea'
export * from './command'
export * from './popover'