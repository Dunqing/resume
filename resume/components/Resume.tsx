import type { ResumeProps } from '@resumejs/components'
import { Resume as ResumeComponent } from '@resumejs/components'
import { ErrorBoundary } from 'react-error-boundary'
import { FallbackComponent } from './FallbackComponent'

function Resume(props: Omit<ResumeProps, 'children'> & { children?: string }) {
  return (
    <ErrorBoundary
      resetKeys={[props.children]}
      FallbackComponent={FallbackComponent}
    >
      <ResumeComponent
        {...props}
        components={{ ...props?.components }}
      >
        {props.children ?? ''}
      </ResumeComponent>
    </ErrorBoundary>
  )
}

export default Resume
