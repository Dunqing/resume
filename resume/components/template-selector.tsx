"use client"

import * as React from "react"
import { useRouter } from "next/navigation"
import { CaretSortIcon, CheckIcon } from "@radix-ui/react-icons"

import { cn } from "@/lib/utils"


import { Preset } from "../data/presets"
import { Button, Command, CommandEmpty, CommandGroup, CommandInput, CommandItem, Popover, PopoverContent, PopoverProps, PopoverTrigger } from './ui'

interface TemplateSelectorProps extends PopoverProps {
  presets: Preset[]
}

export function TemplateSelector({ presets, ...props }: TemplateSelectorProps) {
  const [open, setOpen] = React.useState(false)
  const [selectedPreset, setSelectedPreset] = React.useState<Preset>()
  const router = useRouter()

  return (
    <Popover open={open} onOpenChange={setOpen} {...props}>
      <PopoverTrigger asChild>
        <Button
          variant="outline"
          role="combobox"
          aria-label="Load a preset..."
          aria-expanded={open}
          className="flex-1 justify-between md:max-w-[200px] lg:max-w-[300px]"
        >
          {selectedPreset ? selectedPreset.name : "Load a preset..."}
          <CaretSortIcon className="ml-2 h-4 w-4 shrink-0 opacity-50" />
        </Button>
      </PopoverTrigger>
      <PopoverContent className="w-[300px] p-0">
        <Command>
          <CommandInput placeholder="Search presets..." />
          <CommandEmpty>No presets found.</CommandEmpty>
          <CommandGroup heading="Examples">
            {presets.map((preset) => (
              <CommandItem
                key={preset.id}
                onSelect={() => {
                  setSelectedPreset(preset)
                  setOpen(false)
                }}
              >
                {preset.name}
                <CheckIcon
                  className={cn(
                    "ml-auto h-4 w-4",
                    selectedPreset?.id === preset.id
                      ? "opacity-100"
                      : "opacity-0"
                  )}
                />
              </CommandItem>
            ))}
          </CommandGroup>
          <CommandGroup className="pt-0">
            <CommandItem onSelect={() => router.push("/examples")}>
              More examples
            </CommandItem>
          </CommandGroup>
        </Command>
      </PopoverContent>
    </Popover>
  )
}