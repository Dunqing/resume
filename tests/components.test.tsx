import { describe, expect, it } from 'vitest'
import { Resume } from '@resumejs/components'
import { render } from '@testing-library/react'

const replaceWhitespace = (md: string) => {
  return md.replace(/\n\s*/g, '\n')
}

describe('components', () => {
  it('heading', () => {
    const md = replaceWhitespace(`
      # 张三

      ![头像](https://notion-avatar.vercel.app/api/svg/eyJmYWNlIjo5LCJub3NlIjoxMCwibW91dGgiOjEsImV5ZXMiOjcsImV5ZWJyb3dzIjoxMSwiZ2xhc3NlcyI6MCwiaGFpciI6MTEsImFjY2Vzc29yaWVzIjoxMywiZGV0YWlscyI6MCwiYmVhcmQiOjAsImZsaXAiOjAsImNvbG9yIjoicmdiYSgyNTUsIDAsIDAsIDApIiwic2hhcGUiOiJub25lIn0=)

      - [https://github.com/Dunqing](https://github.com/Dunqing)

      >

      - google@gmail.com
      - 3 年

      >

      - Telegram: @luckingforme
  `)
    const { container } = render(<Resume>{md}</Resume>)
    expect(container.innerHTML).toMatchSnapshot()
  })

  it('toolbox', () => {
    const md = replaceWhitespace(`
    ---
    theme: true
    print: true
    github: 'https://github.com/Dunqing/resume'
    ---
  `)
    const { container } = render(<Resume>{md}</Resume>)
    expect(container.innerHTML).toMatchSnapshot()
  })

  describe('card', () => {
    it('basic', () => {
      const md = replaceWhitespace(`
        ### 公司内部 Cli 开发（独立完成）
  
        | 担任角色 |     项目周期      |
        | :------: | :---------------: |
        |  负责人  | 2020.11 ~ 2021.01 |
      `)
      const { container } = render(<Resume components={{}}>{md}</Resume>)
      expect(container.innerHTML).toMatchSnapshot()
    })

    it('multi row data', () => {
      const md = replaceWhitespace(`
        ### 公司内部 Cli 开发（独立完成）
  
        | 担任角色 |     项目周期      |
        | :------: | :---------------: |
        |  负责人  | 2020.11 ~ 2021.01 |
        |  负责人1  | 2020.11 ~ 2021.01 |
      `)
      const { container } = render(<Resume components={{}}>{md}</Resume>)
      expect(container.innerHTML).toMatchSnapshot()
    })
  })

  it('description', () => {
    const md = replaceWhitespace(`
      ### [swr-taro](https://www.npmjs.com/package/taro-swr)
      
      **业务背景是由于公司的小程序对需要对接口进行管理,(替换原来实现的 useRequest，api 不太友好)。考虑到内部的 H5 使用的是 SWR，统一请求管理库对维护者更加友好，所以决定把 SWR 适配到 Taro 中去**
    `)
    const { container } = render(<Resume components={{}}>{md}</Resume>)
    expect(container.innerHTML).toMatchSnapshot()
  })
})
