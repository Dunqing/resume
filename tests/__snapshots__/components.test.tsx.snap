// Vitest Snapshot v1

exports[`components > card > basic 1`] = `
"<div class=\\"r-resume\\"><div class=\\"font-sans text-base text-dark-400 bg-white dark:text-light-700 dark:bg-dark-400 r-container\\"><div class=\\"border print:border-none border-light-700 md:w-full md:max-w-screen-md xs:w-full mx-auto p-y-5 sm:p-x-10 p-x-4 dark:border-dark-100 relative r-content\\"><div class=\\"absolute right-4 top-2 flex gap-x-1.5 print:hidden r-toolbox\\"></div><h3 class=\\"m-t-2 m-b-1 p-y-0.5 text-base font-semibold r-h3\\" level=\\"3\\">公司内部 Cli 开发（独立完成）</h3>













<div class=\\"p-y-0.5 grid grid-cols-2 gap-y-0.5 r-card\\"><div class=\\"text-sm r-card-item\\"><span align=\\"center\\" class=\\"text-sm font-medium r-card-item-label\\">担任角色</span>：<span class=\\"text-blue-gray-600 dark:text-blue-gray-300 r-card-item-value\\" align=\\"center\\">负责人</span></div><div class=\\"text-sm justify-self-end r-card-item\\"><span align=\\"center\\" class=\\"text-sm font-medium r-card-item-label\\">项目周期</span>：<span class=\\"text-blue-gray-600 dark:text-blue-gray-300 r-card-item-value\\" align=\\"center\\">2020.11 ~ 2021.01</span></div></div></div></div></div>"
`;

exports[`components > card > multi row data 1`] = `
"<div class=\\"r-resume\\"><div class=\\"font-sans text-base text-dark-400 bg-white dark:text-light-700 dark:bg-dark-400 r-container\\"><div class=\\"border print:border-none border-light-700 md:w-full md:max-w-screen-md xs:w-full mx-auto p-y-5 sm:p-x-10 p-x-4 dark:border-dark-100 relative r-content\\"><div class=\\"absolute right-4 top-2 flex gap-x-1.5 print:hidden r-toolbox\\"></div><h3 class=\\"m-t-2 m-b-1 p-y-0.5 text-base font-semibold r-h3\\" level=\\"3\\">公司内部 Cli 开发（独立完成）</h3>

















<div class=\\"p-y-0.5 grid grid-cols-2 gap-y-0.5 r-card\\"><div class=\\"text-sm r-card-item\\"><span align=\\"center\\" class=\\"text-sm font-medium r-card-item-label r-card-item-label\\">担任角色</span>：<span class=\\"text-blue-gray-600 dark:text-blue-gray-300 r-card-item-value\\" align=\\"center\\">负责人</span></div><div class=\\"text-sm justify-self-end r-card-item\\"><span align=\\"center\\" class=\\"text-sm font-medium r-card-item-label r-card-item-label\\">项目周期</span>：<span class=\\"text-blue-gray-600 dark:text-blue-gray-300 r-card-item-value\\" align=\\"center\\">2020.11 ~ 2021.01</span></div></div><div class=\\"p-y-0.5 grid grid-cols-2 gap-y-0.5 r-card\\"><div class=\\"text-sm r-card-item\\"><span align=\\"center\\" class=\\"text-sm font-medium r-card-item-label r-card-item-label\\">担任角色</span>：<span class=\\"text-blue-gray-600 dark:text-blue-gray-300 r-card-item-value\\" align=\\"center\\">负责人1</span></div><div class=\\"text-sm justify-self-end r-card-item\\"><span align=\\"center\\" class=\\"text-sm font-medium r-card-item-label r-card-item-label\\">项目周期</span>：<span class=\\"text-blue-gray-600 dark:text-blue-gray-300 r-card-item-value\\" align=\\"center\\">2020.11 ~ 2021.01</span></div></div></div></div></div>"
`;

exports[`components > card 1`] = `
"<div class=\\"r-resume\\"><div class=\\"font-sans text-base text-dark-400 bg-white dark:text-light-700 dark:bg-dark-400 r-container\\"><div class=\\"border print:border-none border-light-700 md:w-full md:max-w-screen-md xs:w-full mx-auto p-y-5 sm:p-x-10 p-x-4 dark:border-dark-100 relative r-content\\"><div class=\\"absolute right-4 top-2 flex gap-x-1.5 print:hidden r-toolbox\\"></div><h3 class=\\"m-t-2 m-b-1 p-y-0.5 text-base font-semibold r-h3\\" level=\\"3\\">公司内部 Cli 开发（独立完成）</h3>













<div class=\\"p-y-0.5 grid grid-cols-2 gap-y-0.5 r-card\\"><div class=\\"text-sm r-card-item\\"><span align=\\"center\\" class=\\"text-sm font-medium r-card-item-label\\">担任角色</span>：<span class=\\"text-blue-gray-600 dark:text-blue-gray-300 r-card-item-value\\" align=\\"center\\">负责人</span></div><div class=\\"text-sm justify-self-end r-card-item\\"><span align=\\"center\\" class=\\"text-sm font-medium r-card-item-label\\">项目周期</span>：<span class=\\"text-blue-gray-600 dark:text-blue-gray-300 r-card-item-value\\" align=\\"center\\">2020.11 ~ 2021.01</span></div></div></div></div></div>"
`;

exports[`components > description 1`] = `
"<div class=\\"r-resume\\"><div class=\\"font-sans text-base text-dark-400 bg-white dark:text-light-700 dark:bg-dark-400 r-container\\"><div class=\\"border print:border-none border-light-700 md:w-full md:max-w-screen-md xs:w-full mx-auto p-y-5 sm:p-x-10 p-x-4 dark:border-dark-100 relative r-content\\"><div class=\\"absolute right-4 top-2 flex gap-x-1.5 print:hidden r-toolbox\\"></div><h3 class=\\"m-t-2 m-b-1 p-y-0.5 text-base font-semibold r-h3\\" level=\\"3\\"><a class=\\"text-blue-600 underline dark:text-blue-400 r-a\\" href=\\"https://www.npmjs.com/package/taro-swr\\">swr-taro</a></h3>
<p class=\\"text-sm indent r-description\\"><strong class=\\"font-semibold r-strong\\">业务背景是由于公司的小程序对需要对接口进行管理,(替换原来实现的 useRequest，api 不太友好)。考虑到内部的 H5 使用的是 SWR，统一请求管理库对维护者更加友好，所以决定把 SWR 适配到 Taro 中去</strong></p></div></div></div>"
`;

exports[`components > heading 1`] = `
"<div class=\\"r-resume\\"><div class=\\"font-sans text-base text-dark-400 bg-white dark:text-light-700 dark:bg-dark-400 r-container\\"><div class=\\"border print:border-none border-light-700 md:w-full md:max-w-screen-md xs:w-full mx-auto p-y-5 sm:p-x-10 p-x-4 dark:border-dark-100 relative r-content\\"><div class=\\"absolute right-4 top-2 flex gap-x-1.5 print:hidden r-toolbox\\"></div><header class=\\"flex flex-col items-center p-4 r-header\\"><h1 class=\\"font-bold text-3xl r-header-name\\">张三</h1><img src=\\"https://notion-avatar.vercel.app/api/svg/eyJmYWNlIjo5LCJub3NlIjoxMCwibW91dGgiOjEsImV5ZXMiOjcsImV5ZWJyb3dzIjoxMSwiZ2xhc3NlcyI6MCwiaGFpciI6MTEsImFjY2Vzc29yaWVzIjoxMywiZGV0YWlscyI6MCwiYmVhcmQiOjAsImZsaXAiOjAsImNvbG9yIjoicmdiYSgyNTUsIDAsIDAsIDApIiwic2hhcGUiOiJub25lIn0=\\" alt=\\"头像\\" class=\\"w-40 text-3xl r-header-avatar\\"><ul class=\\"list-none flex flex-wrap r-header-row\\">
<li class=\\"sibling:before:content-| sibling:before:p-x-2 sibling:before:text-gray-400 r-header-col\\"><a class=\\"text-blue-600 underline dark:text-blue-400 r-a\\" href=\\"https://github.com/Dunqing\\">https://github.com/Dunqing</a></li>
</ul><ul class=\\"list-none flex flex-wrap r-header-row\\">
<li class=\\"sibling:before:content-| sibling:before:p-x-2 sibling:before:text-gray-400 r-header-col\\"><a class=\\"text-blue-600 underline dark:text-blue-400 r-a\\" href=\\"mailto:google@gmail.com\\">google@gmail.com</a></li>
<li class=\\"sibling:before:content-| sibling:before:p-x-2 sibling:before:text-gray-400 r-header-col\\">3 年</li>
</ul><ul class=\\"list-none flex flex-wrap r-header-row\\">
<li class=\\"sibling:before:content-| sibling:before:p-x-2 sibling:before:text-gray-400 r-header-col\\">Telegram: @luckingforme</li>
</ul></header>


<blockquote class=\\"border-l-0.3rem p-l-2 border-gray-200 m-b-4 text-gray-700 dark:text-gray-400 r-blockquote\\">
</blockquote>

<blockquote class=\\"border-l-0.3rem p-l-2 border-gray-200 m-b-4 text-gray-700 dark:text-gray-400 r-blockquote\\">
</blockquote>
</div></div></div>"
`;

exports[`components > toolbox 1`] = `
"<div class=\\"r-resume\\"><div class=\\"font-sans text-base text-dark-400 bg-white dark:text-light-700 dark:bg-dark-400 r-container\\"><div class=\\"border print:border-none border-light-700 md:w-full md:max-w-screen-md xs:w-full mx-auto p-y-5 sm:p-x-10 p-x-4 dark:border-dark-100 relative r-content\\"><div class=\\"absolute right-4 top-2 flex gap-x-1.5 print:hidden r-toolbox\\"></div><hr class=\\"r-hr\\">
<h2 class=\\"m-t-3 m-b-1 text-xl font-semibold p-y-1 b-b border-dark dark:border-dark-100 r-h2\\" level=\\"2\\">theme: true
print: true
github: '<a class=\\"text-blue-600 underline dark:text-blue-400 r-a\\" href=\\"https://github.com/Dunqing/resume\\">https://github.com/Dunqing/resume</a>'</h2></div></div></div>"
`;
